platforms/nuttx/src/px4/common/usr_hrt.cpp

```diff
@@ -132,12 +137,13 @@ void
 hrt_init(void)
 {
 	boardctl(HRT_REGISTER, (uintptr_t)&g_hrt_client_handle);
-
+#ifdef PX4_USERSPACE_HRT
+	boardctl(HRT_ABSOLUTE_TIME, (uintptr_t)&g_abstime_base);
```

Comment:
I find it confusing that the same IOCTL number returns either the value or a pointer to the value depending on build type ;

It would be cleaner to either have a separate IOCTL for getting the base, or expanding the "HRT_REGISTER" to return both base and the semaphore...

Reply:
Then what is returned by HRT_REGISTER would be implementation defined, but whatever you prefer.